Source/WebKit/NetworkProcess/mac/NetworkConnectionToWebProcessMac.mm

```diff
@@ -105,11 +113,19 @@
 
     RELEASE_LOG(Process, "Launch Services checkin starting, infoDictionary = %{public}@", (__bridge NSDictionary *)infoDictionary.get());
 
-    Boolean ok = _LSApplicationCheckInProxyPtr()(kLSDefaultSessionID, auditToken.auditToken(), infoDictionary.get(), ^(CFDictionaryRef result, CFErrorRef error) {
+    auto block = makeBlockPtr([protectedThis = Ref { *this }, this, auditToken](CFDictionaryRef result, CFErrorRef error) mutable {
         NSDictionary *dictionary = (__bridge NSDictionary *)result;
         RELEASE_LOG(Process, "Launch Services checkin completed, result = %{public}@, error = %{public}@", dictionary, (__bridge NSError *)error);
+
+        callOnMainRunLoop([protectedThis = WTFMove(protectedThis), this, auditToken = WTFMove(auditToken)] mutable {
+            if (m_pendingDisplayName.isEmpty())
+                return;
+            updateActivePages(std::exchange(m_pendingDisplayName, emptyString()), { }, WTFMove(auditToken));
```

Comment:
Should be using `String()`, not `emptyString()`

Reply:
Also, I would have written this like so:
```
if (auto name = std::exchange(m_pendingDisplayName, String()))
    updateActivePages(name, { }, WTFMove(auditToken));
```

Reply:
I tried this, but got a compile error saying that the conversion from String to bool was ambiguous, so I kept as is for now.